core/src/main/java/org/web3j/protocol/core/filters/Filter.java

```diff
@@ -136,12 +137,15 @@ private void pollFilter(EthFilter ethFilter) {
         }
         if (ethLog.hasError()) {
             Error error = ethLog.getError();
+            String message = error.getMessage();
             switch (error.getCode()) {
                 case RpcErrors.FILTER_NOT_FOUND:
                     reinstallFilter();
                     break;
                 default:
-                    throwException(error);
+                    if (Pattern.compile("(?i)\\bfilter\\s+not\\s+found\\b").matcher(message).find())
+                        reinstallFilter();
+                    else throwException(error);
```

Comment:
Add regex to a final static var and also:

```suggestion
                    Pattern.compile("FILTER_NOT_FOUND_PATTERN").matcher(message).find())? 
                    reinstallFilter():throwException(error);
```

Reply:
Hey @gtebrean, should I commit your changes to my branch and then push it in order to get my PR approved? Sorry, but I'm new to the open source world.

Reply:
Add regex to a final static var  as I mentioned and after implement according to my suggestion ....here in this branch is good

Reply:
also run build locally before push

Reply:
Initially I thought your suggestions were already implemented by you in the code, that's why I commited. 
Anyway, thanks for help. I'll make the right modifications and push.

Reply:
Sorry again..

Reply:
@gtebrean Ternary operator can't be used in this case, since [it would be used as a statement, not as an assignment](https://stackoverflow.com/a/51197166/10276248). Eventually it also generates side effects, thus it should be better to leave it in the normal if-else structure.